src/main/scala/com/advancedtelematic/campaigner/client/ResolverClient.scala

```diff
@@ -0,0 +1,37 @@
+package com.advancedtelematic.campaigner.client
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.util.FastFuture
+import akka.stream.Materializer
+import com.advancedtelematic.campaigner.data.Codecs.uriDecoder
+import com.advancedtelematic.campaigner.data.DataType.ExternalUpdateId
+import com.advancedtelematic.libats.codecs.CirceCodecs._
+import com.advancedtelematic.libats.data.DataType.Namespace
+import com.advancedtelematic.libats.messaging_datatype.DataType.DeviceId
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+import io.circe.syntax._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait ResolverClient {
+  def availableUpdatesFor(ns: Namespace, devices: Seq[DeviceId]): Future[Seq[ExternalUpdateId]]
+}
+
+class ResolverHttpClient(uri: Uri)(implicit ec: ExecutionContext, system: ActorSystem, mat: Materializer)
+  extends HttpClient("external_resolver", uri) with ResolverClient {
+
+  private def externalResolverUri(ns: Namespace): Future[Uri] = {
+    val path = uri.path / "api" / "v1" / "namespace_settings" / ns.get
+    val request = HttpRequest(HttpMethods.GET, uri.withPath(path))
+    execHttp[Uri](ns, request)
+  }
+
+  override def availableUpdatesFor(ns: Namespace, devices: Seq[DeviceId]): Future[Seq[ExternalUpdateId]] = {
+    externalResolverUri(ns).flatMap { uri =>
+      val entity = HttpEntity(ContentTypes.`application/json`, devices.asJson.noSpaces)
```

Comment:
Why not `GET /uri?ids=id1,id2,id3`?

Reply:
I didn't know what requests the resolver accepts, so I made this assumption for now.
If that's the request it accepts I'll change it.

Reply:
I assumed it is in our design document, but seems like it is not. My implementation accepts the get as I described

Reply:
those params on URLs will have a shorter size limit, which is fine, but it means we need to account for that when fetching/batching. Or we could just ignore that for now as there will be only a 2 or 3 devices.

Reply:
AFAIK, query params have size limit only in browsers and there is no such restriction on http level

Reply:
It's not only in browsers, the http protocol itself doesn't have a limitation but in practice some clients/libraries do. But anyway this doesn't matter for now.

Reply:
Updated to `GET` instead of `POST`.